In [45]:
import pandas as pd

In [46]:
data=pd.read_csv('processed_local_large_well_3_1b_dfn.csv', float_precision='round_trip')

In [47]:
data.describe()

Id             X             Y             Z         dX  \
count  3.920000e+06  3.920000e+06  3.920000e+06  3.920000e+06  3920000.0   
mean   1.960000e+06  7.137100e+02  1.063970e+03 -4.750000e+02       10.0   
std    1.131607e+06  4.041349e+02  4.041349e+02  5.773431e+02        0.0   
min    0.000000e+00  1.871000e+01  3.689700e+02 -1.470000e+03       10.0   
25%    9.799998e+05  3.662100e+02  7.164700e+02 -9.725000e+02       10.0   
50%    1.960000e+06  7.137100e+02  1.063970e+03 -4.750000e+02       10.0   
75%    2.939999e+06  1.061210e+03  1.411470e+03  2.250000e+01       10.0   
max    3.919999e+06  1.408710e+03  1.758970e+03  5.200000e+02       10.0   

              dY         dZ      Porosity           kii           kjj  \
count  3920000.0  3920000.0  3.920000e+06  3.920000e+06  3.920000e+06   
mean        10.0       10.0  2.577398e-02  3.122488e-15  3.122118e-15   
std          0.0        0.0  4.437936e-02  6.534291e-15  6.534505e-15   
min         10.0       10.0  2.902800e-04  1.050371e-18  9.349565e-19   
25%         10.0       10.0  4.189503e-03  3.296985e-17  3.222510e-17   
50%         10.0       10.0  5.182442e-03  4.602574e-17  4.555071e-17   
75%         10.0       10.0  6.702098e-03  6.885364e-17  6.906129e-17   
max         10.0       10.0  1.200000e-01  1.700000e-14  1.700000e-14   

                kkk         block  
count  3.920000e+06  3.920000e+06  
mean   3.126365e-15  1.818545e+00  
std    6.532680e-15  3.853948e-01  
min    1.428325e-18  1.000000e+00  
25%    3.611656e-17  2.000000e+00  
50%    5.025345e-17  2.000000e+00  
75%    7.486194e-17  2.000000e+00  
max    1.700000e-14  2.000000e+00

In [48]:
min (data.X), max (data.X), data['X'].median()

(18.71, 1408.71, 713.71)

In [49]:
min (data.Y), max (data.Y), data['Y'].median()

(368.97, 1758.97, 1063.97)

In [50]:
min (data.Z), max (data.Z), data['Z'].median()

(-1470, 520, -475.0)

now make a copy and translate the mesh

In [51]:
large_well_3_1b = data.copy(deep=True)

need to manually calculate the translation distance based on injection location and bounds of the new/refine mesh versus the full domain mesh

In [52]:
large_well_3_1b.X = large_well_3_1b.X+36.29

In [53]:
large_well_3_1b.Y = large_well_3_1b.Y+56.03

In [144]:
large_well_3_1b.Z = large_well_3_1b.Z-565

now verify translation was correct, and also chck the bounds to make sure still within the full domain

In [54]:
min (large_well_3_1b.X), max (large_well_3_1b.X), large_well_3_1b['X'].median()

(55.0, 1445.0, 750.0)

In [55]:
min (large_well_3_1b.Y), max (large_well_3_1b.Y), large_well_3_1b['Y'].median()

(425.0, 1815.0, 1120.0)

In [10]:
min (large_well_3_1b.Z), max (large_well_3_1b.Z), large_well_3_1b['Z'].median()

(-1470, 520, -475.0)

Z are out of bounds, min value in full domainstarts at 0 (for the nodes).  min for 10m cell is -1470

fix Z

In [148]:
large_well_3_1b.Z = large_well_3_1b.Z+215

In [149]:
min (large_well_3_1b.Z), max (large_well_3_1b.Z), large_well_3_1b['Z'].median()

(-1470, 520, -475.0)

because the file is so big, it can't be edited in Excel, so need to add some additional columns to the file now, and then later add the 10 row header that is usually written by Leapfrog

need to add column (1) for Id, columns 5, 6, and 7 for dX, dY, and dZ

first add the "Id"

In [17]:
large_well_3_1b.insert(loc=0, column='Id', value=large_well_3_1b.index)

now add the dX, dY, and dZ

In [18]:
large_well_3_1b.insert(loc=4, column='dX', value=10)

In [19]:
large_well_3_1b.insert(loc=5, column='dY', value=10)

In [20]:
large_well_3_1b.insert(loc=6, column='dZ', value=10)

now check the file

In [21]:
large_well_3_1b.head()

Id      X        Y    Z  dX  dY  dZ  Porosity  Compressibility  kii  kjj  \
0   0  18.71  1758.97  520  10  10  10       0.0              0.0  0.0  0.0   
1   1  18.71  1758.97  510  10  10  10       0.0              0.0  0.0  0.0   
2   2  18.71  1758.97  500  10  10  10       0.0              0.0  0.0  0.0   
3   3  18.71  1758.97  490  10  10  10       0.0              0.0  0.0  0.0   
4   4  18.71  1758.97  480  10  10  10       0.0              0.0  0.0  0.0   

   kkk  
0  0.0  
1  0.0  
2  0.0  
3  0.0  
4  0.0

now sort so values are in correct order for em2ex

In [22]:
a = large_well_3_1b.sort_values(["Z","Y","X"], ascending = [True, True, True])

In [23]:
a.head()

Id      X       Y     Z  dX  dY  dZ  Porosity  Compressibility  \
27999    27999  18.71  368.97 -1470  10  10  10  0.001837     1.100548e-08   
55999    55999  28.71  368.97 -1470  10  10  10  0.001397     8.485526e-09   
83999    83999  38.71  368.97 -1470  10  10  10  0.001746     1.087847e-08   
111999  111999  48.71  368.97 -1470  10  10  10  0.002155     1.266467e-08   
139999  139999  58.71  368.97 -1470  10  10  10  0.002349     1.554523e-08   

                 kii           kjj           kkk  
27999   6.472289e-18  1.049329e-17  1.399280e-17  
55999   7.881279e-18  1.211391e-17  8.001830e-18  
83999   1.474609e-17  1.043076e-17  1.335984e-17  
111999  1.573106e-17  2.113084e-17  2.836260e-17  
139999  2.098971e-17  3.076099e-17  3.149767e-17

now we have the perm and proposity all set up, but if the mesh has both granite and sediment, we need to all a column for the block number, and re-assign the perm and porosity for the sediments

first export the block model from Leapfrog, then run global2local.py to get it into local coords, then read it in here

In [24]:
GMdata=pd.read_csv('local_large_3-1b_Block Model_cell.csv', float_precision='round_trip')

In [25]:
b = GMdata.sort_values(["Z","Y","X"], ascending = [True, True, True])

In [26]:
b.head()

Id      X       Y     Z  dX  dY  dZ         GM
0   0  17.91  366.78 -1470  10  10  10  Granitiod
1   1  27.91  366.78 -1470  10  10  10  Granitiod
2   2  37.91  366.78 -1470  10  10  10  Granitiod
3   3  47.91  366.78 -1470  10  10  10  Granitiod
4   4  57.91  366.78 -1470  10  10  10  Granitiod

sediments = block 1, grnit = block 2

In [27]:
b['block'] = 0

In [28]:
b.loc[b.GM == 'Granitiod', 'block'] = 2

In [29]:
b.GM.unique()

array(['Granitiod', 'Valley Fill'], dtype=object)

In [30]:
b.loc[b.GM == 'Valley Fill', 'block'] = 1

In [31]:
b.block.unique()

array([2, 1])

In [32]:
b.head()

Id      X       Y     Z  dX  dY  dZ         GM  block
0   0  17.91  366.78 -1470  10  10  10  Granitiod      2
1   1  27.91  366.78 -1470  10  10  10  Granitiod      2
2   2  37.91  366.78 -1470  10  10  10  Granitiod      2
3   3  47.91  366.78 -1470  10  10  10  Granitiod      2
4   4  57.91  366.78 -1470  10  10  10  Granitiod      2

In [33]:
b.index = a.index

In [34]:
a['block'] = 0

In [35]:
a['block'] = b['block'].values

In [36]:
a.head()

Id      X       Y     Z  dX  dY  dZ  Porosity  Compressibility  \
27999    27999  18.71  368.97 -1470  10  10  10  0.001837     1.100548e-08   
55999    55999  28.71  368.97 -1470  10  10  10  0.001397     8.485526e-09   
83999    83999  38.71  368.97 -1470  10  10  10  0.001746     1.087847e-08   
111999  111999  48.71  368.97 -1470  10  10  10  0.002155     1.266467e-08   
139999  139999  58.71  368.97 -1470  10  10  10  0.002349     1.554523e-08   

                 kii           kjj           kkk  block  
27999   6.472289e-18  1.049329e-17  1.399280e-17      2  
55999   7.881279e-18  1.211391e-17  8.001830e-18      2  
83999   1.474609e-17  1.043076e-17  1.335984e-17      2  
111999  1.573106e-17  2.113084e-17  2.836260e-17      2  
139999  2.098971e-17  3.076099e-17  3.149767e-17      2

In [37]:
a.loc[a.block == 1, 'kii'] = 1.7e-14

In [38]:
a.loc[a.block == 1, 'kjj'] = 1.7e-14

In [39]:
a.loc[a.block == 1, 'kkk'] = 1.7e-14

In [40]:
a.loc[a.block == 1, 'Porosity'] = 0.12

In [41]:
del a['Compressibility']

In [42]:
a.describe()

Id             X             Y             Z         dX  \
count  3.920000e+06  3.920000e+06  3.920000e+06  3.920000e+06  3920000.0   
mean   1.960000e+06  7.137100e+02  1.063970e+03 -4.750000e+02       10.0   
std    1.131607e+06  4.041349e+02  4.041349e+02  5.773431e+02        0.0   
min    0.000000e+00  1.871000e+01  3.689700e+02 -1.470000e+03       10.0   
25%    9.799998e+05  3.662100e+02  7.164700e+02 -9.725000e+02       10.0   
50%    1.960000e+06  7.137100e+02  1.063970e+03 -4.750000e+02       10.0   
75%    2.939999e+06  1.061210e+03  1.411470e+03  2.250000e+01       10.0   
max    3.919999e+06  1.408710e+03  1.758970e+03  5.200000e+02       10.0   

              dY         dZ      Porosity           kii           kjj  \
count  3920000.0  3920000.0  3.920000e+06  3.920000e+06  3.920000e+06   
mean        10.0       10.0  2.577398e-02  3.122488e-15  3.122118e-15   
std          0.0        0.0  4.437936e-02  6.534291e-15  6.534505e-15   
min         10.0       10.0  2.902800e-04  1.050371e-18  9.349565e-19   
25%         10.0       10.0  4.189503e-03  3.296985e-17  3.222510e-17   
50%         10.0       10.0  5.182442e-03  4.602574e-17  4.555071e-17   
75%         10.0       10.0  6.702098e-03  6.885364e-17  6.906129e-17   
max         10.0       10.0  1.200000e-01  1.700000e-14  1.700000e-14   

                kkk         block  
count  3.920000e+06  3.920000e+06  
mean   3.126365e-15  1.818545e+00  
std    6.532680e-15  3.853948e-01  
min    1.428325e-18  1.000000e+00  
25%    3.611656e-17  2.000000e+00  
50%    5.025345e-17  2.000000e+00  
75%    7.486194e-17  2.000000e+00  
max    1.700000e-14  2.000000e+00

now can write the data to a CSV file.

In [43]:
a.to_csv("processed_local_large_well_3_1b_dfn.csv", encoding='utf-8', index=False)

In [44]:
a.describe().to_csv("processed_local_large_well_3_1b_dfn_Summary.csv")